models/tasks.js

```diff
@@ -35,6 +35,28 @@ const updateTask = async (taskData, taskId = null) => {
   }
 }
 
+/**
+ * Updates self task status
+ *
+ * @param taskData { Object }: task data object to be stored in DB
+ * @param taskId { string }: taskid which will be used to update the task in DB
+ * @return {Promise<{taskId: string}>}
+ */
+const updateTaskStatus = async (taskData, taskId) => {
+  try {
+    taskData = await toFirestoreData(taskData)
+    const task = await tasksModel.doc(taskId).get()
+    await tasksModel.doc(taskId).set({
+      ...task.data(),
+      ...taskData
+    })
+    return { taskId }
+  } catch (err) {
+    logger.error('Error in updating task status', err)
+    throw err
+  }
```

Comment:
Tomorrow if we wish to allow them them to change task percentage, are we going to create a new function `updateTaskProgress`? We create a generalised function and then we decide what is to be passed in that function. We already have such function `updateTask`, let's use that only.